torch_frame/data/dataset.py

```diff
@@ -278,7 +303,9 @@ def __call__(
             y = self._get_mapper(self.target_col).forward(
                 df[self.target_col], device=device)
 
-        return TensorFrame(feat_dict, self.col_names_dict, y)
+        tf = TensorFrame(feat_dict, self.col_names_dict, y)
+        self._merge_feat(tf)
+        return tf
```

Comment:
The `_merge_feat` changes the tensor frame inplace. I think it might be faster than to create a new tf object and modifying the reference? What do you think @weihua916

Reply:
Returning the object doesn't necessarily mean that it'll have to create a new object, and in both cases, there's no performance difference as far as I understand correctly. Since I think this is more of a preference, the decision is yours!

Reply:
I think we can return tensor frame object as suggested by @akihironitta It should not affect speed since we are just passing around the reference to the in-place modified object.

Reply:
Okay. I fixed the code.